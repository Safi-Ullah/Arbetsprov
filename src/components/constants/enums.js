export const Enums = {
    TASK_STATUS: {
        created: 1,
        active: 2,
        completed: 3,
        onhold: 4,
        deleted: 5
    },
    TASK_ACTIONS: {
        edit: 1,
        setComplexity: 2,
        delete: 3,
    }
};