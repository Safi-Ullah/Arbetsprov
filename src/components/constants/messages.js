export const PromptMessages = {
    DELETE_TASK_PROMPT: `Are you sure you want to permanently delete this task?
        You will not be able to restore in the future.`
};