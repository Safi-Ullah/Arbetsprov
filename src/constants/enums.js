export const Enums = {
    TASK_STATUS: {
        created: 1,
        active: 2,
        completed: 3,
        onhold: 4,
        deleted: 5
    },
    TASK_ACTIONS: {
        edit: 1,
        setComplexity: 2,
        delete: 3,
    },
    APP_SCREENS: {
        dashboard: 1,
        deletedTasks: 2
    },
    SORT_BY_COMPLEXITY: {
        ascending: 1,
        descending: 2
    },
    TEST_TYPES: {
        dom: 'Dom test',
        snapshot: 'Snapshot test'
    }
};