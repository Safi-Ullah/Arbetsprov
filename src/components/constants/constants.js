import { Enums } from './enums';

export const TASK_ACTION_HANDLERS = {
    [Enums.TASK_ACTIONS.edit]: 'displayEditTaskModal',
    [Enums.TASK_ACTIONS.delete]: 'displayConfirmationDialog'
};