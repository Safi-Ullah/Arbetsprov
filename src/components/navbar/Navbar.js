import React from 'react';
import PropTypes from 'prop-types';
import { Enums } from '../../constants';
import { getUniqueId } from '../../utils';

export const Navbar = ({ addTask, setNewTask,
    newTask, setSelectedScreen }) => <nav className="navbar navbar-dark navbar-expand-lg bg-dark">
        <button
            className="btn btn-link navbar-brand"
            onClick={() => setSelectedScreen(Enums.APP_SCREENS.dashboard)}>
            Arbetsprov
        </button>
        <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation">
            <span className="navbar-toggler-icon" />
        </button>

        <div className="collapse navbar-collapse" id="navbarSupportedContent">
            <ul className="navbar-nav mr-auto">
                <li className="nav-item active">
                    <button
                        className="btn btn-link nav-link"
                        onClick={() => setSelectedScreen(Enums.APP_SCREENS.deletedTasks)}>
                        Deleted Tasks
                    </button>
                </li>
            </ul>
            <form className="form-inline my-2 my-lg-0">
                <input className="form-control mr-sm-2"
                    placeholder="Add Task"
                    value={newTask ? newTask.title : ''}
                    onChange={event => setNewTask({
                        id: getUniqueId(),
                        title: event.target.value,
                        status: Enums.TASK_STATUS.created,
                        complexity: "0"
                    })}
                    aria-label="Add Task" />
                <button className="btn btn-outline-secondary my-2 my-sm-0"
                    onClick={addTask}>
                    Add
                </button>
            </form>
        </div>
    </nav>;

Navbar.propTypes = {
    addTask: PropTypes.func.isRequired,
    setNewTask: PropTypes.func.isRequired,
    newTask: PropTypes.object.isRequired,
    setSelectedScreen: PropTypes.func.isRequired
};