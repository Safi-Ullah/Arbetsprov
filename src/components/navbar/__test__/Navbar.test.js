import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Navbar } from '../Navbar';
import { Enums } from '../../../constants';

const props = {
    addTask: jest.fn(),
    setNewTask: jest.fn(),
    newTask: {},
    setSelectedScreen: jest.fn()
};

describe(Enums.TEST_TYPES.snapshot, () => {
    it('tests snapshot', () => {
        const tree = TestRenderer.create(<Navbar {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});