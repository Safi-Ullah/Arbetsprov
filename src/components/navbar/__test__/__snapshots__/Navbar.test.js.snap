// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test tests snapshot 1`] = `
<nav
  className="navbar navbar-dark navbar-expand-lg bg-dark"
>
  <button
    className="btn btn-link navbar-brand"
    onClick={[Function]}
  >
    Arbetsprov
  </button>
  <button
    aria-controls="navbarSupportedContent"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarSupportedContent"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarSupportedContent"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        className="nav-item active"
      >
        <button
          className="btn btn-link nav-link"
          onClick={[Function]}
        >
          Deleted Tasks
        </button>
      </li>
    </ul>
    <form
      className="form-inline my-2 my-lg-0"
    >
      <input
        aria-label="Add Task"
        className="form-control mr-sm-2"
        onChange={[Function]}
        placeholder="Add Task"
      />
      <button
        className="btn btn-outline-secondary my-2 my-sm-0"
        onClick={[MockFunction]}
      >
        Add
      </button>
    </form>
  </div>
</nav>
`;
